docs/assembly.rst

```diff
@@ -906,8 +940,16 @@ Pseudocode::
 Opcode Stream Generation
 ------------------------
 
-During opcode stream generation, we keep track of the current stack height,
-so that accessing stack variables by name is possible.
+During opcode stream generation, we keep track of the current stack height
+in a counter,
+so that accessing stack variables by name is possible. The stack height is modified with every opcode
+that modifies the stack and with every label that is annotated with a stack
+adjustment. Every time a new
+local variable is introduced, it is registered together with the current
+stack height. If a variable is accessed (either for copying its value or for
```

Comment:
So,
```
e
label[x]:
```
is not generally equal to
```
let a := e
```
.

This means, perhaps, a programmer needs to keep track of two stack heights mentally (I'm still trying to figure out how to read a program written in this language).